llvm/lib/Support/Triple.cpp

```diff
@@ -563,6 +563,15 @@ static Triple::SubArchType parseSubArch(StringRef SubArchName) {
       (SubArchName.endswith("r6el") || SubArchName.endswith("r6")))
     return Triple::MipsSubArch_r6;
 
+  if (SubArchName.startswith("spir")) {
+    if (SubArchName.endswith("fpga"))
+      return Triple::SPIRSubArch_fpga;
+    else if (SubArchName.endswith("x86_64"))
+      return Triple::SPIRSubArch_x86_64;
```

Comment:
AOT compile for x86 should use native triple. Maybe could be added as a separate PR later.

Reply:
Using native triple within -fsycl-targets will require a bit more work to validate as the offload paths for -fsycl are expecting SPIRV.  A separate PR to change would be ideal.